app/services/order_cycle_form.rb

```diff
@@ -1,13 +1,50 @@
-class OrderCycleForm
-  attr_accessor :order_cycle, :params
+require 'open_food_network/permissions'
+require 'open_food_network/proxy_order_syncer'
 
-  def initialize(order_cycle, params)
+class OrderCycleForm
```

Comment:
I'm happy to extract the logic. I'm just wondering about the name `OrderCycleForm` now. It's not really representing a form, it's containing modelling logic.

Reply:
Fair enough. I'll admit that it is really only one half of a standard form object (in that it is not being used to render the form itself), but it is still encapsulates the logic required for creating/updating the underlying model, and is very much coupled to the data provided by the submission of the form.